File: packages/app/src/services/software/definitions/lightshot.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'lightshot';
	public name = 'Lightshot';
	public category = [SoftwareCategory.Media, SoftwareCategory.Utility];
	public downloadName = 'setup-lightshot.exe';
	public icon = 'lightshot.png';
	public homepage = 'https://app.prntscr.com';

	public async resolveDownloadUrl() {
		return ok('https://app.prntscr.com/build/setup-lightshot.exe');
	}
}

File: src/lib/software/lightshot.py
from src.lib.software import BaseSoftware

class Lightshot(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'setup-lightshot.exe'
        self.silent_install_args = ['/VERYSILENT', '/SUPPRESSMSGBOXES', '/NORESTART', '/SP-']

    def resolve_download_url(self):
        self.url_resolved.emit('https://app.prntscr.com/build/setup-lightshot.exe')
